extension/extensionlimiter/extensionlimiter.go

```diff
@@ -0,0 +1,60 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package extensionlimiter // import "go.opentelemetry.io/collector/extension/extensionlimiter"
+
+import (
+	"context"
+)
+
+// Option is passed to limiter providers.
+//
+// NOTE: For data-specific or tenant-specific limits we will extend
+// providers with Options and add a Config type, but none are
+// supported yet and this PR contains only interfaces, not need for
+// options in core repository components.
+type Option interface {
+	apply()
+}
+
+// Checker is for checking when a limit is saturated.  This can be
+// called prior to the start of work to check for limiter saturation.
+type Checker interface {
```

Comment:
Up for discussion (not sure): Should this be the "basic" limiter, then call it "Limiter" instead?

Reply:
I had actually named this `Limiter` in an earlier draft. Yes.